homeassistant/components/mobile_app/__init__.py

```diff
@@ -15,24 +16,46 @@
 
 async def async_setup(hass: HomeAssistantType, config: ConfigType):
     """Set up the mobile app component."""
+    hass.data[DOMAIN] = {DATA_DELETED_IDS: []}
     store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
     app_config = await store.async_load()
     if app_config is None:
-        app_config = {DATA_DELETED_IDS: [], DATA_REGISTRATIONS: {}}
+        app_config = {DATA_DELETED_IDS: []}
 
-    if hass.data.get(DOMAIN) is None:
-        hass.data[DOMAIN] = {DATA_DELETED_IDS: [], DATA_REGISTRATIONS: {}}
+    hass.data[DOMAIN] = {
+        DATA_DELETED_IDS: app_config.get(DATA_DELETED_IDS, []),
+        DATA_STORE: store
+    }
 
-    hass.data[DOMAIN][DATA_DELETED_IDS] = app_config.get(DATA_DELETED_IDS, [])
-    hass.data[DOMAIN][DATA_REGISTRATIONS] = app_config.get(DATA_REGISTRATIONS,
-                                                           {})
-    hass.data[DOMAIN][DATA_STORE] = store
-
-    for registration in app_config[DATA_REGISTRATIONS].values():
-        setup_registration(hass, store, registration)
-
-    register_http_handlers(hass, store)
+    hass.http.register_view(RegistrationsView())
     register_websocket_handlers(hass)
-    register_deleted_webhooks(hass, store)
+    register_deleted_webhooks(hass)
 
     return True
+
+
+async def async_setup_entry(hass, entry):
```

Comment:
For future PR: add support for `async_unload_entry` (unload webhook) and `async_remove_entry` (add entry webhook to deleted IDs)

Reply:
https://github.com/home-assistant/home-assistant/issues/22044